src/main/java/org/spongepowered/common/entity/SpongeEntityArchetypeBuilder.java

```diff
@@ -39,33 +35,37 @@
 import org.spongepowered.api.entity.Entity;
 import org.spongepowered.api.entity.EntityArchetype;
 import org.spongepowered.api.entity.EntityType;
-import org.spongepowered.api.entity.EntityTypes;
-import org.spongepowered.common.data.util.DataUtil;
+import org.spongepowered.common.data.nbt.validation.DelegateDataValidator;
+import org.spongepowered.common.data.nbt.validation.Validations;
 import org.spongepowered.common.util.Constants;
 
+import java.util.Objects;
 import java.util.Optional;
 
 public class SpongeEntityArchetypeBuilder extends AbstractDataBuilder<EntityArchetype> implements EntityArchetype.Builder {
 
-    EntityType entityType = EntityTypes;
+    EntityType entityType = null;
     DataContainer entityData;
     CompoundNBT compound;
+    DataManipulator.Mutable manipulator;
 
     public SpongeEntityArchetypeBuilder() {
         super(EntityArchetype.class, Constants.Sponge.EntityArchetype.BASE_VERSION);
     }
 
     @Override
     public EntityArchetype.Builder reset() {
-        this.entityType = UNKNOWN;
+        this.entityType = null;
         this.entityData = null;
+        this.manipulator = null;
         return this;
     }
 
     @Override
     public EntityArchetype.Builder from(final EntityArchetype value) {
         this.entityType = value.getType();
         this.entityData = value.getEntityData();
+        this.manipulator = null; // TODO get cached values if present
```

Comment:
Honestly shouldn't need to, on reading the entity data if the archetype is queried, it's fine to re-generate the manipulators from nbt. It's difficult to make the nbt "change watched" to invalidate the manipulators otherwise.